xbmc/pvr/PVRManager.cpp

```diff
@@ -1647,18 +1650,21 @@ std::string CPVRManager::GetPlayingTVGroupName()
   return IsStarted() && m_guiInfo ? m_guiInfo->GetPlayingTVGroup() : "";
 }
 
-void CPVRManager::UpdateLastWatched(CPVRChannel &channel)
+void CPVRManager::UpdateLastWatched(const CPVRChannelPtr &channel)
 {
+  if (!channel)
```

Comment:
I know that a ref can't be NULL... this would have segfaulted before `CPVRChannel* foo(NULL); UpdateLastWatched(*foo)`